src/App.tsx

```diff
@@ -120,6 +120,18 @@ function App(): ReactElement {
 
   const [isRecording, setIsRecording] = useState(false);
   const timeControls = useConstructor(() => new TimeControls(canv!, playbackFps));
+  // TODO: Move all logic for the time slider into its own component!
```

Comment:
could timeControls encapsulate this logic?

Reply:
Maybe, but I think the flags I have here are all more interaction-focused, and `TimeControls` is more of a pure utility class. Having a UI component handle these sorts of interaction minutia makes more sense to me.

Reply:
Agree.  And I like the idea of having "business logic" classes that are independent of ui framework, such that the ui framework makes calls into that api. Feels more maintainable when we try to have that separation.